Original Main Updated

willliam9 opened the pull request:
## Agreement
```
By creating a pull request in stk-code, you hereby agree to dual-license your contribution as
GNU General Public License version 3 or any later version and
Mozilla Public License version 2 or any later version.

This includes your previous contribution(s) under the same name of contributor.

Keep the above statement in the pull request comment for agreement.

```

kimden reviewed (approved):
It seems like the changes correspond to the latest 113 commits made into supertuxkart/stk-code:master, starting from Nov 10, 2023. Clearly, I do not mind if they are merged into another repository.

Alayan-stk-2 committed:
Minor code cleanup

- Avoid loading the skin spinners in many different places
- Also remove a leftover printf

Alayan-stk-2 committed:
When changing base theme, keep the theme variant if possible

Alayan-stk-2 committed:
Update the Cartoon Coal skin

Changes by CrystalDaEevee:
- Fix some UI elements that were incorrectly using the base Cartoon variant instead of the correctly contrasted Cartoon Coal variant
- Use a dark table-header to fix contrast issues
- Fix Licenses.txt

AsciiWolf committed:
Update AppData file

Add launchable tag that is now required by many AppStream parsers

Alayan-stk-2 committed:
Fix a crash in debug mode

The assert for spinner getStringValue not returning empty is unnecessary.

CodingJellyfish committed:
Guarantee 2048 bones

kimden committed:
Add an option for -fsanitize=undefined

CodingJellyfish committed:
Merge pull request #5117 from kimden/add-usan

Add an option for -fsanitize=undefined

CodingJellyfish committed:
Merge pull request #5047 from ognevny/improve-mingw-build

improve MinGW build

CodingJellyfish committed:
Revert #5047

Icenowy committed:
Fix exit crash when built for GLES2 on Linux (#5118)

kimden committed:
Fix #5116

Moved the entirety of ParticleEmitter and the objects referring to it under #ifndef SERVER_ONLY

CodingJellyfish committed:
Merge pull request #5122 from kimden/fix5116

Fix #5116

kimden committed:
Separate SQL functions from server lobby

I also attempted to reuse some code that checks if a peer is banned

kimden committed:
Get rid of repeated code, add comments

easySQLQuery was generalized to return output rows if requested,
which allowed to shorten the code for many queries.

The code for binding values to sqlite statement was also repeated
many times. Two auxiliary structures were introduced, so that it's
possible to provide at the same time both those parameters which
require and those which don't require binding, in a single
StringUtils::insertValues() call.

deveee committed:
Don't build stk-editor in linux builder as its data files are already removed from stk assets.

Alayan-stk-2 committed:
Fix #5125

Alayan-stk-2 committed:
Fix selection square theming for new cartoon skins

Alayan-stk-2 committed:
Display the main graphical settings in the benchmark result screen

tobbi committed:
Use relative path for MacOS workflow

CodingJellyfish committed:
Merge pull request #5130 from tobbi/macos_workflows_test

[DRAFT BUT FIXED THE FAILURE] Use relative path for MacOS workflow

mr-c committed:
Upgrade SIMDe to version 0.8.2

Nstelt committed:
Center-align Goal and Progress columns on profile page

CodingJellyfish committed:
Try build

CodingJellyfish committed:
Merge pull request #5131 from mr-c/simde_v0.8.2

Upgrade SIMDe to version 0.8.2

CodingJellyfish committed:
Merge pull request #5124 from kimden/cleanup-1

Separate database-related code from ServerLobby

CodingJellyfish committed:
Merge pull request #5133 from Nstelt/center_align_profile_cols

Center-align Goal and Progress columns on profile page

CodingJellyfish committed:
Fix #5134

deveee committed:
Update gradle and avoid building with too new sdk

deveee committed:
Add hint to allow only landscape orientation on android.

Seems to fix that running in portrait mode bug.

CodingJellyfish committed:
Fix #5127

CodingJellyfish committed:
Fix remaining #5127

CodingJellyfish committed:
Merge pull request #5104 from AsciiWolf/patch-1

Update AppData file

Alayan-stk-2 committed:
Groundwork for a favorite-tracks system

- Save a per-player list of favorite tracks in the players config file
- Read said lists and setup a matching track group
- New logic for the isInGroup function to ensure favorite tracks are correctly detected and displayed in the track selection screen
- Switch the active list of favorite tracks when switching the active player profile
- Clarify some unclear comments

Alayan-stk-2 committed:
Add an UI to edit favorite tracks

The UI works simply by ticking a checkbox in the track and GP screen and then selecting a track icon to switch its favorite status. The checkbox is always unticked by default when entering the screen. Changes are saved in the current player profile.

The display of tracks inside groups is not automatically refreshed.

Alayan-stk-2 committed:
Improve the UI display of favorite tracks

- Immediately refresh the display of tracks inside group after favoriting or unfavoriting a track
- Sort tracks according to favorite status, displaying favorite tracks first
- Add heart as a badge overlay option, and use it to mark favorite tracks
- Remove debug prints

CodingJellyfish committed:
Rewrite favorite system

CodingJellyfish committed:
Fix typo & Add easter egg badge

kimden committed:
Fix #4997 (#5149)

In startSelection, assets are erased only after the spectators are determined.
If a spectator doesn't have the picked map, it's not loaded, but the spectator
mode stays afterwards.

Alayan-stk-2 committed:
Fix #5085 and #5152

- In the pause menu, the buttons are part of ribbon widgets. Under certain conditions, the game could incorrectly select the underlying back button as active instead of its ribbon. Because the icon buttons that constitute a ribbon widget don't have their x or y property set, a check that x or y are not negative ensures that keyboard navigation doesn't select them.
- The multiplayer focus dispatcher button is made invisible if there is only one player in the kart selection screen.
- Add a getName() function to the Widget class, to help when debugging GUI.

nyllet committed:
Fix some compiler warnings in irrlicht (#5144)

- Avoid dereferencing null pointer
- Remove some unused variables

Alayan-stk-2 committed:
Improve contrast in the coal themes

Classic Coal:
- Make the default spinner background color slightly darker

Cartoon Coal:
- Make the spinner fill darker than the default spinner background color, instead of lighter
- Make the background color of spinners, buttons, and focused checkboxes somewhat darker

Alayan-stk-2 committed:
Improve things for #5115

Grey backgrounds are fundamentally problematic when trying to display colored texts. This commit improves perceived contrast, but compromises are made to not change the style of the skins too much.

Classic coal:
- Make the background slightly less transparent and slightly darker
- Make the neutral list blue darker and the red lighter

Cartoon coal:
- Make the neutral list blue darker, the neutral red lighter, the focused blue slightly darker and the focused red darker

kimden committed:
Fix #5138 (#5154)

Alayan-stk-2 committed:
Add a display options cartoon icon

This is simply a copy of the screen_other icon, but it's much better than using the default classic theme icon.

Alayan-stk-2 committed:
Fix #5086

- Allow a track to be separately added to the set of 'battle arena' groups and 'soccer arena' groups
- Update the track removal routine to properly cleanup if a track is present in multiple sets of groups
- Use a new enum and split functions to avoid duplicating code
- Fix a crash introduced in 7f78b5b when a track present in a custom group is not available.

CodingJellyfish committed:
Update upload artifavt version (#5168)

* Update upload artifavt version

* Update download artifavt version

CodingJellyfish committed:
Add STK blog news list to online screen (#5165)

* Rewrite NewsManager

* Add open link function

* Small fixes

* Try fix build crash

* Try fix build crash #2

* Try fix build crash #3

* Fix multiple error message & Add empty indicator

Hope the build would pass since the actions in the pr isn't updated yet

CodingJellyfish committed:
Change skin font reload logic

CodingJellyfish committed:
Add search bar for gp editor

AsciiWolf committed:
Rename AppStream metadata file to a rDNS format (#5137)

Also:
- Set the component type to desktop-application instead of a deprecated desktop one
- Update data/po/update_desktop_file_appdata.py + add the previously added launchable type there

deveee committed:
Update linux build script for updated dependencies

deveee committed:
Make generate assets working on macos

deveee committed:
Avoid unhandled exception on iOS

ektor5 committed:
Add benchmark mode to commandline options (#5178)

Adding `--benchmark` to commandline options to automatically start
benchmark mode, set unlimited FPS, save result and exit.

Signed-off-by: Ettore Chimenti <ettore.chimenti@linaro.org>
Co-authored-by: Ettore Chimenti <ettore.chimenti@linaro.org>

CodingJellyfish committed:
Implement kart grouping/searching/favorite (#5172)

* Implement kart grouping/searching/favprite

* Add search bar for arenas

* Use methods instead of friend class

EYH0602 committed:
doc: add instructions to build on Windows using LLVM MinGW from powershell/cmd (#5180)

CodingJellyfish committed:
Update install instructions

CodingJellyfish committed:
Fix #4964

CodingJellyfish committed:
Prevent mx fps write in command line benchmarking

CodingJellyfish committed:
Fix random kart related issue

CodingJellyfish committed:
Capitalize kart classes

spageektti committed:
Fix typos in `INSTALL.md` (#5196)

CodingJellyfish committed:
Fix #4971 (#5195)

* DPI aware on Windows

* Fix duplicate manifest

* Try fix MSVC

* Try fix MSVC

* Try fix MSVC

andryunin committed:
Update mojoAL, fixes #5188 (#5194)

CodingJellyfish committed:
Let speedometer show score in battles & Delete the 10 player limit in arenas (#5193)

* Let speedometer show score in battles (Fix #5084 & #3832)

* Use red-green fade for ffa

* Fix networking bugs

CodingJellyfish committed:
Fix duplicate tracks in online battle

CodingJellyfish committed:
Fix #4902 (#5191)

CodingJellyfish committed:
Fix crash race result screen on networking

deveee committed:
Small fixes in stkgui files

deveee committed:
Fixed running update_pot script.

Probably needed because of new python.

deveee committed:
Update pot file

deveee committed:
Update translations

CodingJellyfish committed:
Update Changelog (#5199)

* Update CHANGELOG

* Update CHANGELOG again

deveee committed:
Try disable tutorial on android TV to make google happy

qwertychouskie committed:
graphical_restrictions.xml: Remove restrictions for Android Emulator (#5170)

Android Emulator supports proper hardware acceleration nowadays, we shouldn't be forcing the legacy pipeline if it's not needed.

CodingJellyfish committed:
Fix missed credit

CodingJellyfish committed:
Fix memory leak

CodingJellyfish committed:
Avoid iterating appear animation node when it's invisible (#5200)

CodingJellyfish committed:
Add favorite to network track & Fix Android favorite kart (#5202)

deveee committed:
Fixed a crash when trying to edit favorites in multiplayer screen.

Also fixed some compiler warnings.

CodingJellyfish committed:
Maybe fix 5226

CodingJellyfish committed:
Fix invalid memory access

CodingJellyfish committed:
Fix Windows armv7 build (#5224)

Since latest Windows SDK dropped armv7 support, we skip the MSVC windows armv7 build on GitHub actions. There's still armv7 in llvm-mingw.

kimden committed:
Move ranking related stuff out of server_lobby.cpp (#5223)

FifthTundraG committed:
Add Arch install instructions and fix typos in INSTALL.md (#5217)

Alayan-stk-2 committed:
Fully update the Changelog

- Add the items that were missing in the 1.4 changelog
- Update the 1.5 changelog with recent changes and a few that were missing
- Try to improve clarity for multiple 1.5 entries

Alayan-stk-2 committed:
Fix #5173

Do not modify the max-players-in-game parameter to interpret it in server_config.cpp, instead do the interpretation in server_lobby.cpp

The issue was caused by the modified parameter being then saved in the XML, causing issues later on.

This commit also addresses a similar issue with the min-start-game-players parameter

Alayan-stk-2 committed:
Disable unusable in-game option tabs (fix #5163)

Alayan-stk-2 committed:
Improvements in the last lap music transition

- Group up the code to trigger the fast music and the last lap SFX in the same area
- Remove all the code meant to play a music at higher pitch, it was unused and would have been annoying if it was.
- Apply the 'last lap SFX' reduced music volume when switching between a normal music and a fast music too
- Increase the switch duration between normal and fast music from 1.0s to 1.5s

Alayan-stk-2 committed:
Miscellaneous improvements

- Add more colors to indicate the time difference with a ghost kart, with new thresholds at +- 0.3s and +-2.5s, while keeping noticeable contrast between colors
- Confirm in the networking documentation that STK's networking uses UDP

Alayan-stk-2 committed:
Fix #5096

- Move the GP higshcore update from race_result_gui.cpp to world.cpp ; GUI code should not be doing this sort of operation
- Update the GP hishcore for any number of eligible local human players.
- Various simplifications of the highscore logic in world.cpp
- Drop support for lap-trial GP highscore - the code for it was extremely hacky and a fix would require a complete rewrite of how lap totals are tracked in lap-trial GPs

Alayan-stk-2 committed:
Fix #5204

When music was resumed, its status was switched to 'playing' without regard to the music play initialization steps having been done or not.

Alayan-stk-2 committed:
Fix #5232

- Don't clear the favorites data of the kart properties manager when leaving the kart selection screen - this was the cause of the bug.
- Remove a block of code that was never accessed because the condition was always false
- Use getIdent() behind the scenes for more robustness
- Prevent a crash when uninstalling an addon that's registered as a favorite kart

DesiOtaku committed:
Command line parameters to assign keyboard or gamepad with the -N or -R option (#5230)

Authored-by: Tej A. Shah (DesiOtaku)

Alayan-stk-2 committed:
Improve command-line input device handling

- Fallback to the first keyboard config instead of crashing if the requested device doesn't exist
- Automatically enable the requested config if it's available but disabled

kimden committed:
Make all deltas in GP online result screen dependent on server points (#5235)

Alayan-stk-2 committed:
Race Result GUI improvements

- Consolidate duplicated per-team code in the CTF player score drawing
- Don't assume the value of the extra wait time after race results have been displayed
- Tweak timings for the GP UI, in particular increase the duration of points transfer from new to total from 1.0s to 2.0s
- Use a lambda for a minor code quality improvement

Alayan-stk-2 committed:
Fix #5226 and some other GUI issues

- Ensure the back button will always work when clicked in the arena screen
- Fix the back button not working in the online track selection screen by removing the ability to edit favorites in the network track selection screen (it was poor UX anyway). The ability to see previously picked favorites remains.
- Avoid the spinners of the random GP screen overflowing with big fonts in non-lap-trial mode

Alayan-stk-2 committed:
Fix #5014

The TextBoxWidget relies under the hood on IGUIEditBox and behaves in a peculiar way. How exactly both the 'send' button and the chat box could be simultaneously active is unclear beyond involving use of both keyboard and mouse, and the underlying issue will likely remain until a deeper overhaul.

In the meantime, simply selecting back the chatbox after the send box has been triggered ensures that at most one line will be affected, and is convenient in other situations where the send box is used.

Alayan-stk-2 committed:
Introduce super-sampling options

Allow to set the render scale over 100%, with presets at 125%, 150% and 200%. Users with powerful GPUs can therefore get a higher quality image at the cost of performance. Render scale increase the vertical resolution and horizontal resolution by the scaling factor, 125% is therefore 1.56x more pixels than native, 150% is 2.25x more and 200% is 4x more pixels.

kimden committed:
Remove unused variables, fix #5161 (#5241)

Previously some of them were commented in one file without realizing they are extern in another file.

deveee committed:
Add shaderc deps initialization to the android build script.

deveee committed:
Initialize shaderc deps for linux build script in the same way as for android.

CodingJellyfish committed:
Fix memory corruption with shared wheel and headlight, see #5240

MollyMaclachlan committed:
Removed glow under main menu options, see #5109 (#5245)

kimden committed:
Minor fixes for database initialization scripts (#5248)

CodingJellyfish committed:
Make SSAO correct (#5249)

* Removed glow under main menu options, see #5109

* Much better SSAO

* Fix shader bug

* Increase kernel size

* Increase kernel size & bring back pute shader

* Reduce sample to 4

* Fix black line

* Stronger bias when resolution is lower

Alayan-stk-2 committed:
Reorganize graphics settings, add 4096 shadows

- Add 4096 resolution shadows. This helps noticeably in some places to reduce shadow aliasing, nonetheless, for most users, the significant performance hit will not be worth the small quality gains.
- Add an 8th graphics preset, with max resolution shadows.
- Enable Light Shaft (God Rays) only on level 5, instead of level 4
- Enable SSAO only on level 7 instead of level 6
- Changed the location of specific settings in the custom settings dialog, to group the blur options at the end and to have less performance-intensive options towards the top. The options activated on each preset level are now 'connected'.
- Remove the text saying the game needs to be restarted to apply new settings - the player doesn't need to manually restart the game, STK takes care of it when needed
- Remove unused variables and fix typo

Alayan-stk-2 committed:
Fix #5270

- Snap all normal camera back to their default position when starting or restarting a race, irrespective of the previous camera status
- When a player stops looking backward, snap the forward camera to the correct orientation.

Because the smoothing camera modifies the real distance at which the kart is looked at, the camera will move away from the kart at the start of a race, but this movement is a lot more natural than the one that existed previously.

Tested to also work in splitscreen.

Alayan-stk-2 committed:
Direct users to the GitHub in the crash message

Alayan-stk-2 committed:
Discourage farming in ranked

- Improve the data accuracy logic: the upperbound and lowerbound logic failed to reduce point exchanges with low rated players with a much higher RD, which encouraged 'farming'.
- Increase the base points per second from 0.18 to 0.25.
- Update comments.

As a result of these changes, ratings will change quicker when playing against players of broadly similar skill (within 1500 points) and slower against players of a very different level.

kimden committed:
Fix #5121: Check if chat messages start with 'name: ' (#5254)

* Check if chat messages start with 'name: '

* Check one username only

Alayan-stk-2 committed:
Improve camera behavior

- Ensure that the distance the smooth camera stays from the kart is much closer to what the parameters specify when the kart is standing still. When moving, the camera will still change its distance depending on speed, but in a range closer to the specified value than previously.
- When snapping the camera back, properly set angle and distance. This improves things again for #5270
- Also snap the camera when moving the kart in the overworld by clicking challenges on the minimap

Known issues:
- The ground lighting associated with skidding makes an incorrect assumption about the distance at which it can be cut off. This only matters with small camera distances.

Alayan-stk-2 committed:
Fix #5278

Simply don't move the camera when rewinding.

Alayan-stk-2 committed:
Update default camera settings

- Make the height of the smoothed camera change slightly with speed, to stay closer to the requested angle
- Update the standard camera default settings to not be too close to the kart. These settings are inspired by my own custom camera settings and many previous discussions about a tweak to the standard camera values.
- Fix an unrelated assert that had not been properly updated.

Alayan-stk-2 committed:
Add logic to update the standard camera settings when upgrading

Because the defaults in the code get written into config.xml,use a special parameter to ensure that the standard camera settings are updated for people switching from an older 1.x version.

yerlotic committed:
Fix ImageMagick 7 deprecation warning (#5291)

* Fix imagemagick 7 deprecation warning

* Use magick if it's available

Alayan-stk-2 committed:
Fix #5290

Alayan-stk-2 committed:
Fix #5283

Avoid keyboard navigation moving back to the same widget if possible, as in some situations skipping over invisible or disabled widgets could loop back to the starting point and prevent from accessing parts of the screen.

CodingJellyfish committed:
Fixed undeleted text box widget if pressing esc to exit

Semphriss committed:
Animate the parachute and bubblegum shield (#5306)

Co-authored-by: Your Name <you@example.com>

CodingJellyfish committed:
Try to fix #5207

m_icons_frame texture pointer should be retrieved again after sameRestart

CodingJellyfish committed:
Implement PCSS & Revert graphics level 8 (#5300)

* Avoid iterating appear animation node when it's invisible

* Support Ghost Replay Info

* Implement PCSS & Revert graphics level 8

* Add missing file

* Order the effects & etc.

* Use fixed kernel & tune shadow height & remove translation string

* Improve shadow bias

* Tune

Alayan-stk-2 committed:
Add a separate Camera tab

The User Interface tab was crowded, display didn't fit very well either, and a new tab allows to lay out camera options more directly.

The graphics settings icon is now used for the camera tab. The cartoon theme has a new icon for the graphics tab, the standard theme is using the old icon as a placeholder.

Alayan-stk-2 committed:
Move the content of the camera dialog into the camera tab

maksalees committed:
Put the macOS icon in a rounded rectangle (#5299)

CodingJellyfish committed:
Reduce uniform size

CodingJellyfish committed:
Slow down new animation a bit for low fps

CodingJellyfish committed:
Fix #5301

limburgher committed:
Add include to work with gcc15. (#5310)

kimden committed:
Use the addon date format for news screen (#5312)

Also add a function that converts day/month/year, and not TimeType, to date string

CodingJellyfish committed:
Fix PCSS Bugs

CodingJellyfish committed:
Fix #5212

CodingJellyfish committed:
Fix #5208

kimden committed:
Add Georgian font (#5313)

scootergrisen committed:
Change Anvil to Anchor and Bowling to Bowling ball in artist debug menu (#5327)

* Change Anvil to Anchor in artist debug menu

* Change "Bowling" to "Bowling ball"

trofi committed:
src/network/protocols/ranking.hpp: add missing `<stdint.h>` include (#5346)

Without the chnage the build fails on upcoming `gcc-15` as:

    /build/source/src/network/protocols/ranking.hpp:31:5: error: 'uint32_t' does not name a type
       31 |     uint32_t online_id;
          |     ^~~~~~~~

CodingJellyfish committed:
Add option to disable PCSS

CodingJellyfish committed:
Fix #5279

CodingJellyfish committed:
Fix broken defect

CodingJellyfish committed:
Fix build errors

CodingJellyfish committed:
Restrict max karts in arenas

lokpro committed:
fix supertuxkart #5345 (#5360)

CodingJellyfish committed:
Add smooth strength adjustion & Unlock modifying default camera & Fix #5270 (#5365)

* Revert recent camera change & fix #5270

* I tried my best

* Now the bug is properly fixed

* Pull from master

* add back files

* Revert separate camera tab, and move it to display

* Add smooth strength

* Tune

CodingJellyfish committed:
Fix explosion camera bug

CodingJellyfish committed:
Fix out-of-focus text box widget when alphabet characters are assigned as navigation

CodingJellyfish committed:
Fix #5252 & #5189 (#5367)

CodingJellyfish committed:
Make sure only game master controls kart filter (#5368)

CodingJellyfish committed:
Fix memory corruption when closing

CodingJellyfish committed:
Initial work on PBR vulkan

CodingJellyfish committed:
Fix missing PBR textures when material is shared in vulkan engine

CodingJellyfish committed:
Straightforwardly fix #5157

iamnytik committed:
Update cutscene_gui.cpp (#5374)

* Update cutscene_gui.cpp

This update will move the subtitles from the bottom of the screen a few pixels above.This is intended so that the subtitles stay above the Skip buttons in cutscenes

* Update cutscene_gui.cpp

lowered the positioning of subtitles by modifying 10 to 6 font size

deveee committed:
Use vsync on android.

It solves the issue when game rendering just hangs after a while on phones with game mode.

deveee committed:
Disable 1000 fps and 200% scale nonsense on mobile

CodingJellyfish committed:
Disable ghost version check & Disable ghost skid sound

CodingJellyfish committed:
Add warning to oversampling

CodingJellyfish committed:
Proper fix for lighting according to d339d9a

CodingJellyfish committed:
Move srgb255ToLinear to GE

deveee committed:
Use 100% scale on good enough devices

CodingJellyfish committed:
Use srgb255ToLinear for GE PBR if needed

CodingJellyfish committed:
Don't add skid mark quads too frequently which breaks fade-out effect

CodingJellyfish committed:
Properly reset vsync setting (default on) for everyone

CodingJellyfish committed:
Move vertex and index buffer updates into separate conditions

CodingJellyfish committed:
Fix crash when re-entering options screen

CodingJellyfish committed:
Fix more crash

CodingJellyfish committed:
Add initSamplers to SP for reloading samplers live

CodingJellyfish committed:
Make SPTexture reload work with texture compression when max size changed

CodingJellyfish committed:
Remove m_max_size in all ge textures

CodingJellyfish committed:
Clean up Material::getMaskImageMani

CodingJellyfish committed:
Fix OptionsScreenVideo::setImageQuality for reloading live

Only handle mesh texture reloading for now

CodingJellyfish committed:
Erase inserted texture in m_ondemand_load_texture_paths one by one

CodingJellyfish committed:
Ignore PBR data in kart textures preloading

kimden committed:
Make loading screen icons fit ~40% of it (#5383)

I don't know how to get the number of icons in advance, nor I can make proper animations, so for now the jump in size is like that.
I will add more commits if I find out how.

CodingJellyfish committed:
Shader Fix Bundle (#5375)

* That's hell a lot of fix

* More fixes

deveee committed:
More vulkan work

Alayan-stk-2 committed:
Update the changelog and credits

- Put Jellyfish in a separate tab due to his substantial number of commits for 1.5
- Add all major changes since the last changelog update
- Remove a duplicate changelog entry
- Add some missing information about new tracks and track updates in old releases

Alayan-stk-2 committed:
Fix #5318

- Remove a useless loop
- Prevent selecting highscore entries in the track info screen and high score info dialog

deveee committed:
Enable vulkan synchronization validation

Alayan-stk-2 committed:
Avoid duplicate names when the active player is not the first

See issue #5247

deveee committed:
Fix vulkan synchronization issues

From https://github.com/SaschaWillems/Vulkan/commit/ba8fea35ac27f02b62b4f1aa93ae0c3262ee8852
and https://github.com/Overv/VulkanTutorial/commit/dc02dccc6253ebec5d031debceee583f71109bb5

kimden committed:
Don't apply ranking 64 times (left from testing) (#5393)

Note from maintainer: regression originally introduced in commit 3dbf360

deveee committed:
Fix pbr with rotated preTransform

deveee committed:
Remove duplicated code in GEVulkanSkyBoxRenderer

deveee committed:
Add GEVulkanAttachmentTexture

Use VK_IMAGE_USAGE_TRANSIENT_ATTACHMENT_BIT for depth texture, from
https://docs.vulkan.org/samples/latest/samples/performance/render_passes/README.html

deveee committed:
Make push constants reproducible

deveee committed:
Enable depth prepass if necessary

willliam9 committed:
Add a clickable link to the stk website in the credits page (#5398)

Fix #5259

deveee committed:
Remove duplicated pbr code in vulkan shaders

Alayan-stk-2 committed:
Improve the credits code

- Make the credits code more organized
- Add more timing variables instead of relying on obscure code tricks
- Add spacing between the website button and the screen border

Alayan-stk-2 committed:
Update changelog and credits

- Add relevant assets change to the 1.5 changelog ('Oasis' and 'Hole Drop' were not present on 1.4's release and have been moved accordingly)
- Improve structure of some older changelog entries
- Add the initial overworld release to the changelog
- Update credits for contributions in 1.5

Alayan-stk-2 committed:
Fix #5251

Also, when unchecking the 'compare replay' box, the replay that will not be shown is hidden from the list to provide greater clarity on which of the two replays remain active after disabling the comparison. If checking the box back, the second replay will show up again.

Alayan-stk-2 committed:
Improve the ghost replay dialog

- Move the info bubble to be above the control checkboxes
- Don't display the info bubble if the active replay's info field is empty

deveee committed:
Fix possible exit crash

deveee committed:
Fix missing leak sanitizer report

Alayan-stk-2 committed:
Remove duplicated code

Tabs and vertical tab ribbons can use the same creation code to add children elements

Alayan-stk-2 committed:
Improve the code for tab dimensions

- Avoid duplicating a lot of code between tabs and vertical tabs
- Add a few enhancements to tab logic that were present for vertical tabs:
-- Make the font smaller if there is not enough height for a split message (fix text overflow issues in the highscore tabs)
-- Avoid an icon taking more than half the tab's width

deveee committed:
Add asynchronous skybox loading

deveee committed:
Add unicolor array texture

It has VK_IMAGE_USAGE_STORAGE_BIT so can be used in a compute shader

deveee committed:
Restore previous sampler use in mesh texture descriptor

deveee committed:
Make it possible to reload shaders in vulkan

deveee committed:
Fix gles ibl shader

deveee committed:
Fixed some crashes in performance test

Alayan-stk-2 committed:
Fix #5288

- When navigating vertically from a wide widget, prefer navigating to left-aligned widgets than to centered widgets.
- Rename the offsets variable to be more logical

deveee committed:
Add ibl for vulkan

deveee committed:
Improve vulkan pipeline settings

Alayan-stk-2 committed:
Greatly improve the layout of multi-row ribbons

The old formula assigned way too much importance on area, the new one limits it to little more than a tiebreak factor, and better weighs the number of displayed icons and the resulting icon sizes.

Unfortunately, the overall behavior is still complex, so changing slightly the resolution doesn't always produce the expected result. Nonetheless, the results are much better across the board.

- Move the score computations inside the function that used to compute icon-button properties.
- Consider not only having icons at full height, but also icons at a reduced height. This sometimes allow to display all needed elements without having to use another row and even smaller icons. This is especially important for ribbons with a small number of icons, such as the game mode ribbon.
- Take into account the readability margin between columns when computing how many elements can be shown
- Don't pretend that more icons can't fit within a row if icons are below the XML target icon width ; there is already a penalty for layouts with icons lacking height
- If the ribbon has a label, don't count the label area as part of the total area to cover with icons
- Introduce a penalty for not showing the target number of items, to greatly reduce near-misses where the ribbon is scrollable because of a tiny number of elements.
- Transmit the computed target size to the individual icon-buttons, so that margins between buttons can be preserved.
- Use a const value to define the side-margin target
- Don't penalize icons that are above the target size defined in XML
- Minor adjustments to code layout and removal of a few useless lines

Alayan-stk-2 committed:
Improve the layout of screens using dynamic ribbons

This works together with the previous commit to produce as good results as possible. Most notably, the game mode ribbon doesn't require scrolling at any sensible resolution, and the track selection screen stops being always short of showing all standard tracks by a handful, again removing the need for scrolling. People with many addon karts and tracks are also able to see a greater number at once, provided the game screen size is big enough.

- Remove the hint on the number of karts in the kart selection screen, in favor of using the real number of karts
- Update the hint for game mode selection to target 8 items, as there are currently 8 game modes available
- Remove the hint on the number of tracks in the track selection screen, in favor of using the real number of tracks
- Other minor tweaks and adjustments to the GUI files, such as picking icon size targets that produce better results

fiftydinar committed:
Add `StartupWMClass` to the desktop file (#5404)

This resolves some edge-cases where desktop icon isn't shown, like in Gnome's dash panel for example.

Alayan-stk-2 committed:
Add a font to render Hindi

Alayan-stk-2 committed:
Update translations

Alayan-stk-2 committed:
Update the changelog

deveee committed:
Apply bullet patch to support backface culling

From https://github.com/bulletphysics/bullet3/commit/c2f9d927c2349ac91ec9fedce0a2cb45e8c3b82f

deveee committed:
Apply bullet patch to fix undefined behaviour

From https://github.com/bulletphysics/bullet3/commit/0d3940b8bd46cb32408ab626285491acf906b429

deveee committed:
More undefined behaviour fixes

deveee committed:
Fix assertion when using favourite karts uninitialized

deveee committed:
Add point and spot lights rendering for vulkan

deveee committed:
Enable kart lighting effects on vulkan

deveee committed:
Update for latest MoltenVK

deveee committed:
Add spot lights rendering for opengl

deveee committed:
Fix vulkan skinning shader

v_tangent.w is bitangent sign

Alayan-stk-2 committed:
Fix #5408

Alayan-stk-2 committed:
Fix #5411

For the instant speed increase to work, the game applies a speed floor that gets reset the next frame.

This speed floor is now also adjusted by the adjustSpeed() function.

Alayan-stk-2 committed:
Fix #5417

Alayan-stk-2 committed:
Fix #5419

- Increase the character limit for the name before forced scrolling from 15 to 30. Beyond 30, many parts of the GUI become problematic anyways, and while many reasonable usernames exceed 15, none exceed 30.
- Use the actual rendered width of the player name to compute how many characters can be shown at once if scrolling is needed.
-- Also check the width of substrings to avoid overflows from uneven character lengths
- Add a short pause at the beginning and the end of the name scrolling, as otherwise there is little time to see the first and last characters compared to others

Hanuko33 committed:
Add a random track button outside of the track chooser (#5399)

Signed-off-by: Piotr Danecki <i3riced@mailfence.com>

Alayan-stk-2 committed:
Adjust the Tracks and GPs screen to better fit the random track button

Also fix the portability issue from uint introduced in 14b356a

qwertychouskie committed:
Update mojoAL to latest version, fixes #5418 (#5421)

Alayan-stk-2 committed:
Fix #5420

This only applies to normal fonts, not to title fonts

Alayan-stk-2 committed:
Make spinner widgets use the small font size if the label doesn't fit

Alayan-stk-2 committed:
Remove the random track option within the tracks ribbon widget

It's not needed anymore now that the screen has a dedicated random track button

Alayan-stk-2 committed:
Tweaks to the kart and track selection screens

Mostly to make it harder for labels to overflow

deveee committed:
Load spot lights data from exporter settings

deveee committed:
Optimize spot lights shaders

deveee committed:
Fix server only build

Alayan-stk-2 committed:
Fix #5424

Alayan-stk-2 committed:
Allow single-row ribbons to be smaller to show more elements

- Reuse the scoring and resizing logic developped for multi-ribbon widgets
- Set a cap so that the smallest resize factor is 0.7
- Slightly recenter vertically the icons (avoids having a lot of empty space below while scrapping the ribbon border above)

Alayan-stk-2 committed:
Account more accurately for scrolling arrows size in dynamic ribbon widgets

It ensures that the defined spacing between icons is respected when the scrolling arrows are huge, and when the scrolling arrows are quite small, it frees up some additional space to make icons larger.

deveee committed:
Try to fix switch build

deveee committed:
Another try to fix switch build

deveee committed:
Fixed cmake 4.0 warnings

deveee committed:
Fixed DEBUG flag for Visual Studio and also add missing libraries for MinGW build.

deveee committed:
Try to fix macos build

Alayan-stk-2 committed:
Fix #5429

- Don't show tips after game modes without relevant tips
- Only show tips relevant to time-trial after a time-trial race
- Ensure both tips about powerups during races and general driving tips from time-trial tips are shown in racing modes with powerups

Alayan-stk-2 committed:
Improve IconButton behavior and fix #5413

- Make the font size smaller for icon buttons with word wrap if the text is set to not fit in 2 lines (accounting for text break logic)
- Reserve more width for the race result GUI buttonbar (75% instead of 65%)
- Make the icon-buttons in the buttonbar use all of the reserved width instead of being limited to a square shape. This allows to properly use available width at wide resolutions without introducing issues at tall resolutions.
- Also fix a minor comparison issue in the previous commit

Alayan-stk-2 committed:
Fix server-only build

SvenAndreasBelting committed:
Updated Cartoon focus icons.

Main menu and online icons are brighter and have a different border now, to increase visual difference to the normal icons.

deveee committed:
Add RISC-V to linux build script + some small fixes

Alayan-stk-2 committed:
Improve textbox highlight for the cartoon coal skin
